src/utils/compact.ts

```diff
@@ -0,0 +1,17 @@
+type FalseyVals = '' | 0 | false | null | undefined;
```

Comment:
`NaN` is also [considered a falsy value](https://developer.mozilla.org/en-US/docs/Glossary/Falsy). I might be missing something, but should we also include it in this list?